lib/utils/patterns.ts

```diff
@@ -359,10 +359,31 @@ export function inferOrSelectDevice(preferredUuid: string) {
 	});
 }
 
+async function getApplicationByIdOrName(
+	sdk: BalenaSdk.BalenaSDK,
+	idOrName: string,
+) {
+	if (validation.looksLikeInteger(idOrName)) {
+		try {
+			return await sdk.models.application.get(Number(idOrName));
+		} catch (error) {
+			const { BalenaApplicationNotFound } = await import('balena-errors');
+			if (!(error instanceof BalenaApplicationNotFound)) {
```

Comment:
There is a function called `instanceOf` in `lib/errors.ts` that may be safer to use than plain `instanceof` in this case, in the event that dependencies (including `balena-sdk`) end up importing a different version of `balena-errors` (because of incompatible requirements in each dependency's own `package.json` file). (In the past, this led to bugs reported by users.)
```js
    const { instanceOf } = await import('../errors');
    const { BalenaApplicationNotFound } = await import('balena-errors');
    if (!(instanceOf(error, BalenaApplicationNotFound))) {
        ...
```
See https://github.com/balena-io/balena-cli/blob/v12.14.13/lib/errors.ts#L65

Reply:
Updated.